llvm/lib/Target/DirectX/DXILIntrinsicExpansion.cpp

```diff
@@ -67,11 +67,44 @@ static bool isIntrinsicExpansion(Function &F) {
   case Intrinsic::dx_sign:
   case Intrinsic::dx_step:
   case Intrinsic::dx_radians:
+  case Intrinsic::vector_reduce_add:
+  case Intrinsic::vector_reduce_fadd:
     return true;
   }
   return false;
 }
 
+static Value *expandVecReduceFAdd(CallInst *Orig) {
+  // Note: vector_reduce_fadd first argument is a starting value
+  // Our use doesn't need it, so ignoring argument zero.
```

Comment:
Why not write it to handle the starting value, and if it's a constant zero, then the first add is folded/skipped?

Reply:
This behavior is tested via tests with the  `*_start1` postfix.